src/main/scala/edu/berkeley/cs/rise/opaque/Utils.scala

```diff
@@ -1112,22 +1128,25 @@ object Utils extends Logging {
   }
 
   def serializeAggOp(
-    groupingExpressions: Seq[Expression],
-    aggExpressions: Seq[NamedExpression],
+    groupingExpressions: Seq[NamedExpression],
+    aggExpressions: Seq[AggregateExpression],
     input: Seq[Attribute]): Array[Byte] = {
-    // aggExpressions contains both grouping expressions and AggregateExpressions. Transform the
-    // grouping expressions into AggregateExpressions that collect the first seen value.
```

Comment:
Maybe it's helpful to preserve the part of this comment that's still relevant, and perhaps augment it with some more context? Something like "The output of agg operator contains both the grouping columns and the aggregate values. To avoid the need for special handling of the grouping columns, we transform the grouping expressions into AggregateExpressions that collect the first seen value."

Reply:
Added, thanks!